File: DevExpress-Demo/CS/DockingDemo.Wpf/DemoControls.cs
using DevExpress.Xpf.Editors;
using System.Windows.Controls;

namespace DockingDemo {
    public class CodeViewPresenter : DevExpress.Xpf.DemoBase.Helpers.Internal.CodeViewPresenter { }
    public class AutoScrollableTextEdit : TextEdit {
        public AutoScrollableTextEdit() {
            EditValueChanged += AutoScrollableTextEdit_EditValueChanged;
        }
        private void AutoScrollableTextEdit_EditValueChanged(object sender, EditValueChangedEventArgs e) {
            Dispatcher.BeginInvoke(new System.Action(() => {
                ScrollToEnd();
            }));
        }
        public void ScrollToEnd() {
            if(EditCore != null)
                ((TextBox)this.EditCore).ScrollToEnd();
        }
    }
}

File: DevExpress-Demo/VB/DockingDemo.Wpf/DemoControls.vb
Imports DevExpress.Xpf.Editors
Imports System.Windows.Controls

Namespace DockingDemo

    Public Class CodeViewPresenter
        Inherits DevExpress.Xpf.DemoBase.Helpers.Internal.CodeViewPresenter

    End Class

    Public Class AutoScrollableTextEdit
        Inherits TextEdit

        Public Sub New()
            AddHandler EditValueChanged, AddressOf AutoScrollableTextEdit_EditValueChanged
        End Sub

        Private Sub AutoScrollableTextEdit_EditValueChanged(ByVal sender As Object, ByVal e As EditValueChangedEventArgs)
            Dispatcher.BeginInvoke(New System.Action(Sub() ScrollToEnd()))
        End Sub

        Public Sub ScrollToEnd()
            If EditCore IsNot Nothing Then CType(EditCore, TextBox).ScrollToEnd()
        End Sub
    End Class
End Namespace
